In [1]:
# import libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use('ggplot')
pd.set_option('display.max_columns', 200) 

In [2]:
df = pd.read_csv('AB_NYC_2019.csv', sep=',').copy()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [3]:
df.head().T

0  \
id                                                            2539   
name                            Clean & quiet apt home by the park   
host_id                                                       2787   
host_name                                                     John   
neighbourhood_group                                       Brooklyn   
neighbourhood                                           Kensington   
latitude                                                  40.64749   
longitude                                                -73.97237   
room_type                                             Private room   
price                                                          149   
minimum_nights                                                   1   
number_of_reviews                                                9   
last_review                                             2018-10-19   
reviews_per_month                                             0.21   
calculated_host_listings_count                                   6   
availability_365                                               365   

                                                    1  \
id                                               2595   
name                            Skylit Midtown Castle   
host_id                                          2845   
host_name                                    Jennifer   
neighbourhood_group                         Manhattan   
neighbourhood                                 Midtown   
latitude                                     40.75362   
longitude                                   -73.98377   
room_type                             Entire home/apt   
price                                             225   
minimum_nights                                      1   
number_of_reviews                                  45   
last_review                                2019-05-21   
reviews_per_month                                0.38   
calculated_host_listings_count                      2   
availability_365                                  355   

                                                                  2  \
id                                                             3647   
name                            THE VILLAGE OF HARLEM....NEW YORK !   
host_id                                                        4632   
host_name                                                 Elisabeth   
neighbourhood_group                                       Manhattan   
neighbourhood                                                Harlem   
latitude                                                   40.80902   
longitude                                                  -73.9419   
room_type                                              Private room   
price                                                           150   
minimum_nights                                                    3   
number_of_reviews                                                 0   
last_review                                                     NaN   
reviews_per_month                                               NaN   
calculated_host_listings_count                                    1   
availability_365                                                365   

                                                              3  \
id                                                         3831   
name                            Cozy Entire Floor of Brownstone   
host_id                                                    4869   
host_name                                           LisaRoxanne   
neighbourhood_group                                    Brooklyn   
neighbourhood                                      Clinton Hill   
latitude                                               40.68514   
longitude                                             -73.95976   
room_type                                       Entire home/apt   
price                     

In [4]:
# drop unnecessary columns

df.drop(columns='host_name', inplace=True)

In [5]:
# rename columns

df.rename(columns={
    'id': 'listing_id',
    'name': 'listing_name',
    'number_of_reviews': 'total_reviews',
    'last_review': 'last_review_date',
    'calculated_host_listings_count': 'host_total_listings',
    'availability_365': 'available_days_in_year',
    }, inplace=True)

df.tail()

listing_id                                       listing_name  \
48890    36484665    Charming one bedroom - newly renovated rowhouse   
48891    36485057      Affordable room in Bushwick/East Williamsburg   
48892    36485431            Sunny Studio at Historical Neighborhood   
48893    36485609               43rd St. Time Square-cozy single bed   
48894    36487245  Trendy duplex in the very heart of Hell's Kitchen   

        host_id neighbourhood_group       neighbourhood  latitude  longitude  \
48890   8232441            Brooklyn  Bedford-Stuyvesant  40.67853  -73.94995   
48891   6570630            Brooklyn            Bushwick  40.70184  -73.93317   
48892  23492952           Manhattan              Harlem  40.81475  -73.94867   
48893  30985759           Manhattan      Hell's Kitchen  40.75751  -73.99112   
48894  68119814           Manhattan      Hell's Kitchen  40.76404  -73.98933   

             room_type  price  minimum_nights  total_reviews last_review_date  \
48890     Private room     70               2              0              NaN   
48891     Private room     40               4              0              NaN   
48892  Entire home/apt    115              10              0              NaN   
48893      Shared room     55               1              0              NaN   
48894     Private room     90               7              0              NaN   

       reviews_per_month  host_total_listings  available_days_in_year  
48890                NaN                    2                       9  
48891                NaN                    2                      36  
48892                NaN                    1                      27  
48893                NaN                    6                       2  
48894                NaN                    1                      23

In [6]:
df = df[['listing_id', 'listing_name', 
         'available_days_in_year',
        'neighbourhood', 'neighbourhood_group', 
        'room_type', 'price','minimum_nights', 
        'total_reviews', 'last_review_date', 'reviews_per_month',
        'latitude', 'longitude',
        'host_id', 'host_total_listings',        
        ]]

df.head()

listing_id                                      listing_name  \
0        2539                Clean & quiet apt home by the park   
1        2595                             Skylit Midtown Castle   
2        3647               THE VILLAGE OF HARLEM....NEW YORK !   
3        3831                   Cozy Entire Floor of Brownstone   
4        5022  Entire Apt: Spacious Studio/Loft by central park   

   available_days_in_year neighbourhood neighbourhood_group        room_type  \
0                     365    Kensington            Brooklyn     Private room   
1                     355       Midtown           Manhattan  Entire home/apt   
2                     365        Harlem           Manhattan     Private room   
3                     194  Clinton Hill            Brooklyn  Entire home/apt   
4                       0   East Harlem           Manhattan  Entire home/apt   

   price  minimum_nights  total_reviews last_review_date  reviews_per_month  \
0    149               1              9       2018-10-19               0.21   
1    225               1             45       2019-05-21               0.38   
2    150               3              0              NaN                NaN   
3     89               1            270       2019-07-05               4.64   
4     80              10              9       2018-11-19               0.10   

   latitude  longitude  host_id  host_total_listings  
0  40.64749  -73.97237     2787                    6  
1  40.75362  -73.98377     2845                    2  
2  40.80902  -73.94190     4632                    1  
3  40.68514  -73.95976     4869                    1  
4  40.79851  -73.94399     7192                    1

* review columns, values and data types

In [7]:
df.describe(include='all').T

count unique              top   freq  \
listing_id              48895.0    NaN              NaN    NaN   
listing_name              48879  47905   Hillside Hotel     18   
available_days_in_year  48895.0    NaN              NaN    NaN   
neighbourhood             48895    221     Williamsburg   3920   
neighbourhood_group       48895      5        Manhattan  21661   
room_type                 48895      3  Entire home/apt  25409   
price                   48895.0    NaN              NaN    NaN   
minimum_nights          48895.0    NaN              NaN    NaN   
total_reviews           48895.0    NaN              NaN    NaN   
last_review_date          38843   1764       2019-06-23   1413   
reviews_per_month       38843.0    NaN              NaN    NaN   
latitude                48895.0    NaN              NaN    NaN   
longitude               48895.0    NaN              NaN    NaN   
host_id                 48895.0    NaN              NaN    NaN   
host_total_listings     48895.0    NaN              NaN    NaN   

                                  mean              std       min        25%  \
listing_id              19017143.23618   10983108.38561    2539.0  9471945.0   
listing_name                       NaN              NaN       NaN        NaN   
available_days_in_year      112.781327       131.622289       0.0        0.0   
neighbourhood                      NaN              NaN       NaN        NaN   
neighbourhood_group                NaN              NaN       NaN        NaN   
room_type                          NaN              NaN       NaN        NaN   
price                       152.720687        240.15417       0.0       69.0   
minimum_nights                7.029962         20.51055       1.0        1.0   
total_reviews                23.274466        44.550582       0.0        1.0   
last_review_date                   NaN              NaN       NaN        NaN   
reviews_per_month             1.373221         1.680442      0.01       0.19   
latitude                     40.728949          0.05453  40.49979    40.6901   
longitude                    -73.95217         0.046157 -74.24442  -73.98307   
host_id                 67620010.64661  78610967.032667    2438.0  7822033.0   
host_total_listings           7.143982        32.952519       1.0        1.0   

                               50%          75%          max  
listing_id              19677284.0   29152178.5   36487245.0  
listing_name                   NaN          NaN          NaN  
available_days_in_year        45.0        227.0        365.0  
neighbourhood                  NaN          NaN          NaN  
neighbourhood_group            NaN          NaN          NaN  
room_type                      NaN          NaN          NaN  
price                        106.0        175.0      10000.0  
minimum_nights                 3.0          5.0       1250.0  
total_reviews                  5.0         24.0        629.0  
last_review_date               NaN          NaN          NaN  
reviews_per_month             0.72         2.02         58.5  
latitude                  40.72307    40.763115     40.91306  
longitude                -73.95568   -73.936275    -73.71299  
host_id                 30793816.0  107434423.0  274321313.0  
host_total_listings            1.0          2.0        327.0

#### first insights:

* listing_id values are unique for each rows but unique listing_name are less
* average available days of a property is 45 days in a year
* minimum_nights has some extremely high values up to 1250
* host_total_listings max value is 327

In [8]:
df.duplicated().sum()

0

In [9]:
df.isna().sum()

listing_id                    0
listing_name                 16
available_days_in_year        0
neighbourhood                 0
neighbourhood_group           0
room_type                     0
price                         0
minimum_nights                0
total_reviews                 0
last_review_date          10052
reviews_per_month         10052
latitude                      0
longitude                     0
host_id                       0
host_total_listings           0
dtype: int64

In [10]:
# view nan values in listing_name column

df_filtered = df[df['listing_name'].isna()]
df_filtered.head(25).T

2854             3703               5775   \
listing_id                        1615764          2232600            4209595   
listing_name                          NaN              NaN                NaN   
available_days_in_year                362              341                  0   
neighbourhood           Battery Park City     East Village  Greenwich Village   
neighbourhood_group             Manhattan        Manhattan          Manhattan   
room_type                 Entire home/apt  Entire home/apt    Entire home/apt   
price                                 400              200                225   
minimum_nights                       1000                1                  1   
total_reviews                           0               28                  1   
last_review_date                      NaN       2015-06-08         2015-01-01   
reviews_per_month                     NaN             0.45               0.02   
latitude                         40.71239         40.73215           40.73473   
longitude                        -74.0162        -73.98821          -73.99244   
host_id                           6676776         11395220           20700823   
host_total_listings                     1                1                  1   

                                  5975          6269          6567   \
listing_id                      4370230       4581788       4756856   
listing_name                        NaN           NaN           NaN   
available_days_in_year                0             0             0   
neighbourhood                    Nolita  Williamsburg      Bushwick   
neighbourhood_group           Manhattan      Brooklyn      Brooklyn   
room_type               Entire home/apt  Private room  Private room   
price                               215           150            70   
minimum_nights                        7             1             1   
total_reviews                         5             0             0   
last_review_date             2016-01-02           NaN           NaN   
reviews_per_month                  0.09           NaN           NaN   
latitude                       40.72046       40.7137      40.70046   
longitude                      -73.9955     -73.94378     -73.92825   
host_id                        22686810      21600904       1832442   
host_total_listings                   1             1             1   

                                     6605          8841             11963  \
listing_id                         4774658       6782407          9325951   
listing_name                           NaN           NaN              NaN   
available_days_in_year                   0             0                0   
neighbourhood           Washington Heights  Williamsburg   Hell's Kitchen   
neighbourhood_group              Manhattan      Brooklyn        Manhattan   
room_type                     Private room  Private room  Entire home/apt   
price                                   40            45              190   
minimum_nights                           1             1                4   
total_reviews                            0             0                1   
last_review_date                       NaN           NaN       2016-01-05   
reviews_per_month                      NaN           NaN             0.02   
latitude                          40.85198      40.71354         40.76436   
longitude                        -73.93108     -73.93882        -73.98573   
host_id                           24625694      31147528         33377685   
host_total_listings                      1             1                1   

                                  12824         13059         13401  \
listing_id                      9787590       9885866      10052289   
listing_name                        NaN           NaN           NaN   
available_days_in_year                0             0           362   
neighbourhood                    Harlem     Chinatown   Brownsville   
neighbourhood_

* they represent real rooms only with missing names, these may not have been booked during the year
* for these rows, I will assign a default name such as 'Untitled Listing'

In [11]:
# replace missing listing names with 'Untitled Listing'

df['listing_name'].fillna('Untitled Listing', inplace=True)

In [12]:
# view 0 values in available_days_in_year column

df_filtered = df[df['available_days_in_year'] == 0]
df_filtered.head(25).T

4    \
listing_id                                                          5022   
listing_name            Entire Apt: Spacious Studio/Loft by central park   
available_days_in_year                                                 0   
neighbourhood                                                East Harlem   
neighbourhood_group                                            Manhattan   
room_type                                                Entire home/apt   
price                                                                 80   
minimum_nights                                                        10   
total_reviews                                                          9   
last_review_date                                              2018-11-19   
reviews_per_month                                                    0.1   
latitude                                                        40.79851   
longitude                                                      -73.94399   
host_id                                                             7192   
host_total_listings                                                    1   

                                       6    \
listing_id                            5121   
listing_name               BlissArtsSpace!   
available_days_in_year                   0   
neighbourhood           Bedford-Stuyvesant   
neighbourhood_group               Brooklyn   
room_type                     Private room   
price                                   60   
minimum_nights                          45   
total_reviews                           49   
last_review_date                2017-10-05   
reviews_per_month                      0.4   
latitude                          40.68688   
longitude                        -73.95596   
host_id                               7356   
host_total_listings                      1   

                                                       8    \
listing_id                                            5203   
listing_name            Cozy Clean Guest Room - Family Apt   
available_days_in_year                                   0   
neighbourhood                              Upper West Side   
neighbourhood_group                              Manhattan   
room_type                                     Private room   
price                                                   79   
minimum_nights                                           2   
total_reviews                                          118   
last_review_date                                2017-07-21   
reviews_per_month                                     0.99   
latitude                                          40.80178   
longitude                                        -73.96723   
host_id                                               7490   
host_total_listings                                      1   

                                                  14   \
listing_id                                       6090   
listing_name            West Village Nest - Superhost   
available_days_in_year                              0   
neighbourhood                            West Village   
neighbourhood_group                         Manhattan   
room_type                             Entire home/apt   
price                                             120   
minimum_nights                                     90   
total_reviews                                      27   
last_review_date                           2018-10-31   
reviews_per_month                                0.22   
latitude                                      40.7353   
longitude                                   -74.00525   
host_id                                         11975   
host_total_listings                                 1   

                                                     20   \
listing_id                                          7801   
listing_name            Sweet and Spacious Brooklyn Loft   
available_days_in_year       

* even they may represent valid records that were simply unavailable during the last year, they should be removed to avoid calculation issues and skewed results 

In [13]:
df = df[df['available_days_in_year'] != 0].reset_index(drop=True)

In [15]:
# view nan values in last_review_date column

df_filtered = df[df['last_review_date'].isna()]
df_filtered.head(25).T

2    \
listing_id                                             3647   
listing_name            THE VILLAGE OF HARLEM....NEW YORK !   
available_days_in_year                                  365   
neighbourhood                                        Harlem   
neighbourhood_group                               Manhattan   
room_type                                      Private room   
price                                                   150   
minimum_nights                                            3   
total_reviews                                             0   
last_review_date                                        NaN   
reviews_per_month                                       NaN   
latitude                                           40.80902   
longitude                                          -73.9419   
host_id                                                4632   
host_total_listings                                       1   

                                                      15   \
listing_id                                           7750   
listing_name            Huge 2 BR Upper East  Cental Park   
available_days_in_year                                249   
neighbourhood                                 East Harlem   
neighbourhood_group                             Manhattan   
room_type                                 Entire home/apt   
price                                                 190   
minimum_nights                                          7   
total_reviews                                           0   
last_review_date                                      NaN   
reviews_per_month                                     NaN   
latitude                                         40.79685   
longitude                                       -73.94872   
host_id                                             17985   
host_total_listings                                     2   

                                                30   \
listing_id                                    11452   
listing_name            Clean and Quiet in Brooklyn   
available_days_in_year                          365   
neighbourhood                    Bedford-Stuyvesant   
neighbourhood_group                        Brooklyn   
room_type                              Private room   
price                                            35   
minimum_nights                                   60   
total_reviews                                     0   
last_review_date                                NaN   
reviews_per_month                               NaN   
latitude                                   40.68876   
longitude                                 -73.94312   
host_id                                        7355   
host_total_listings                               1   

                                              32   \
listing_id                                  11943   
listing_name            Country space in the city   
available_days_in_year                        365   
neighbourhood                            Flatbush   
neighbourhood_group                      Brooklyn   
room_type                            Private room   
price                                         150   
minimum_nights                                  1   
total_reviews                                   0   
last_review_date                              NaN   
reviews_per_month                             NaN   
latitude                                 40.63702   
longitude                               -73.96327   
host_id                                     45445   
host_total_listings                             1   

                                                        182           230  \
listing_id                                            54466         63588   
listing_name            Beautiful Uptown Manhattan apartmnt           LL3   
available_days_in_year                                  365           349   
neighbourhood                   

In [16]:
# check if the missing values in both columns are in the same rows

if df['last_review_date'].isna().equals(df['reviews_per_month'].isna()):
    print("yes they are same.")
else:
    print("nope, check it")


yes they are same.


* these records should be kept because they may represent new or low-traffic rooms that were available but simply did not receive any reviews or even bookings

In [17]:
df.dtypes

listing_id                  int64
listing_name               object
available_days_in_year      int64
neighbourhood              object
neighbourhood_group        object
room_type                  object
price                       int64
minimum_nights              int64
total_reviews               int64
last_review_date           object
reviews_per_month         float64
latitude                  float64
longitude                 float64
host_id                     int64
host_total_listings         int64
dtype: object

In [18]:
df.nunique()

listing_id                31362
listing_name              30886
available_days_in_year      365
neighbourhood               218
neighbourhood_group           5
room_type                     3
price                       645
minimum_nights               99
total_reviews               388
last_review_date           1216
reviews_per_month           929
latitude                  16038
longitude                 13367
host_id                   21809
host_total_listings          47
dtype: int64

In [19]:
# convert 'neighbourhood' and 'room_type' to category

df['neighbourhood'] = df['neighbourhood'].astype('category')
df['room_type'] = df['room_type'].astype('category')

In [20]:
df.isna().sum()

listing_id                   0
listing_name                 0
available_days_in_year       0
neighbourhood                0
neighbourhood_group          0
room_type                    0
price                        0
minimum_nights               0
total_reviews                0
last_review_date          5207
reviews_per_month         5207
latitude                     0
longitude                    0
host_id                      0
host_total_listings          0
dtype: int64

In [32]:
df.describe(include='all').T

count unique                 top   freq  \
listing_id              31354.0    NaN                 NaN    NaN   
listing_name              31354  30878      Hillside Hotel     18   
available_days_in_year  31354.0    NaN                 NaN    NaN   
neighbourhood             31354    218  Bedford-Stuyvesant   2478   
neighbourhood_group       31354      5           Manhattan  13559   
room_type                 31354      3     Entire home/apt  16532   
price                   31354.0    NaN                 NaN    NaN   
minimum_nights          31354.0    NaN                 NaN    NaN   
total_reviews           31354.0    NaN                 NaN    NaN   
last_review_date          26147   1216          2019-06-23   1322   
reviews_per_month       26147.0    NaN                 NaN    NaN   
latitude                31354.0    NaN                 NaN    NaN   
longitude               31354.0    NaN                 NaN    NaN   
host_id                 31354.0    NaN                 NaN    NaN   
host_total_listings     31354.0    NaN                 NaN    NaN   

                                   mean              std       min  \
listing_id              21004934.644256  11487770.474874    2539.0   
listing_name                        NaN              NaN       NaN   
available_days_in_year       175.834439       126.195374       1.0   
neighbourhood                       NaN              NaN       NaN   
neighbourhood_group                 NaN              NaN       NaN   
room_type                           NaN              NaN       NaN   
price                        162.091822        254.44475      10.0   
minimum_nights                 8.338426        23.838961       1.0   
total_reviews                 31.849365        51.638559       0.0   
last_review_date                    NaN              NaN       NaN   
reviews_per_month              1.810063         1.805657      0.01   
latitude                      40.728406         0.056562  40.49979   
longitude                    -73.948753           0.0514 -74.24442   
host_id                 81450707.380972  86739413.236775    2571.0   
host_total_listings            10.33546        40.705488       1.0   

                                25%         50%          75%          max  
listing_id              11673538.75  22557498.5  31194515.25   36487245.0  
listing_name                    NaN         NaN          NaN          NaN  
available_days_in_year         55.0       168.0        305.0        365.0  
neighbourhood                   NaN         NaN          NaN          NaN  
neighbourhood_group             NaN         NaN          NaN          NaN  
room_type                       NaN         NaN          NaN          NaN  
price                          70.0       112.0        189.0      10000.0  
minimum_nights                  2.0         3.0          5.0       1250.0  
total_reviews                   2.0        10.0         39.0        629.0  
last_review_date                NaN         NaN          NaN          NaN  
reviews_per_month              0.45        1.23         2.68         58.5  
latitude                  40.688392   40.723725    40.763037     40.91306  
longitude                 -73.98328  -73.954115   -73.930152    -73.71299  
host_id                   8619003.5  41089746.5  145236816.0  274321313.0  
host_total_listings             1.0         1.0          3.0        327.0

In [24]:
# review price column with 0 values 

df_filtered = df[df['price'] == 0]
df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8 entries, 12376 to 14400
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   listing_id              8 non-null      int64   
 1   listing_name            8 non-null      object  
 2   available_days_in_year  8 non-null      int64   
 3   neighbourhood           8 non-null      category
 4   neighbourhood_group     8 non-null      object  
 5   room_type               8 non-null      category
 6   price                   8 non-null      int64   
 7   minimum_nights          8 non-null      int64   
 8   total_reviews           8 non-null      int64   
 9   last_review_date        8 non-null      object  
 10  reviews_per_month       8 non-null      float64 
 11  latitude                8 non-null      float64 
 12  longitude               8 non-null      float64 
 13  host_id                 8 non-null      int64   
 14  host_total_listings     8 n

In [25]:
df_filtered.head(10).T

12376  \
listing_id                                                       18750597   
listing_name            Huge Brooklyn Brownstone Living, Close to it all.   
available_days_in_year                                                 28   
neighbourhood                                          Bedford-Stuyvesant   
neighbourhood_group                                              Brooklyn   
room_type                                                    Private room   
price                                                                   0   
minimum_nights                                                          4   
total_reviews                                                           1   
last_review_date                                               2018-01-06   
reviews_per_month                                                    0.05   
latitude                                                         40.69023   
longitude                                                       -73.95428   
host_id                                                           8993084   
host_total_listings                                                     4   

                                                                13565  \
listing_id                                                   20333471   
listing_name            ★Hostel Style Room | Ideal Traveling Buddies★   
available_days_in_year                                            127   
neighbourhood                                         East Morrisania   
neighbourhood_group                                             Bronx   
room_type                                                Private room   
price                                                               0   
minimum_nights                                                      2   
total_reviews                                                      55   
last_review_date                                           2019-06-24   
reviews_per_month                                                2.56   
latitude                                                     40.83296   
longitude                                                   -73.88668   
host_id                                                     131697576   
host_total_listings                                                 4   

                                                                13745  \
listing_id                                                   20624541   
listing_name            Modern apartment in the heart of Williamsburg   
available_days_in_year                                             73   
neighbourhood                                            Williamsburg   
neighbourhood_group                                          Brooklyn   
room_type                                             Entire home/apt   
price                                                               0   
minimum_nights                                                      5   
total_reviews                                                       3   
last_review_date                                           2018-01-02   
reviews_per_month                                                0.15   
latitude                                                     40.70838   
longitude                                                   -73.94645   
host_id                                                      10132166   
host_total_listings                                                 1   

                                                                    13754  \
listing_id                                                       20639628   
listing_name            Spacious comfortable master bedroom with nice ...   
available_days_in_year                                                176   
neighbourhood                                          Bedford-Stuyvesant   
neighbourhood_group                                              Brooklyn   
room_type                          

In [26]:
df = df[df['price'] != 0].reset_index(drop=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31354 entries, 0 to 31353
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   listing_id              31354 non-null  int64   
 1   listing_name            31354 non-null  object  
 2   available_days_in_year  31354 non-null  int64   
 3   neighbourhood           31354 non-null  category
 4   neighbourhood_group     31354 non-null  object  
 5   room_type               31354 non-null  category
 6   price                   31354 non-null  int64   
 7   minimum_nights          31354 non-null  int64   
 8   total_reviews           31354 non-null  int64   
 9   last_review_date        26147 non-null  object  
 10  reviews_per_month       26147 non-null  float64 
 11  latitude                31354 non-null  float64 
 12  longitude               31354 non-null  float64 
 13  host_id                 31354 non-null  int64   
 14  host_total_listings   

In [28]:
# review total_reviews column with 0 values 

df_filtered = df[df['total_reviews'] == 0]
df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5207 entries, 2 to 31353
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   listing_id              5207 non-null   int64   
 1   listing_name            5207 non-null   object  
 2   available_days_in_year  5207 non-null   int64   
 3   neighbourhood           5207 non-null   category
 4   neighbourhood_group     5207 non-null   object  
 5   room_type               5207 non-null   category
 6   price                   5207 non-null   int64   
 7   minimum_nights          5207 non-null   int64   
 8   total_reviews           5207 non-null   int64   
 9   last_review_date        0 non-null      object  
 10  reviews_per_month       0 non-null      float64 
 11  latitude                5207 non-null   float64 
 12  longitude               5207 non-null   float64 
 13  host_id                 5207 non-null   int64   
 14  host_total_listings     5207

In [31]:
df_filtered.tail(20).T

31334  \
listing_id                           36478343   
listing_name            Welcome all as family   
available_days_in_year                    180   
neighbourhood                     East Harlem   
neighbourhood_group                 Manhattan   
room_type                        Private room   
price                                     140   
minimum_nights                              1   
total_reviews                               0   
last_review_date                          NaN   
reviews_per_month                         NaN   
latitude                             40.78749   
longitude                           -73.94749   
host_id                             274273284   
host_total_listings                         1   

                                                                  31335  \
listing_id                                                     36478357   
listing_name            Cozy, Air-Conditioned Private Bedroom in Harlem   
available_days_in_year                                               26   
neighbourhood                                                    Harlem   
neighbourhood_group                                           Manhattan   
room_type                                                  Private room   
price                                                                60   
minimum_nights                                                        1   
total_reviews                                                         0   
last_review_date                                                    NaN   
reviews_per_month                                                   NaN   
latitude                                                       40.80953   
longitude                                                      -73.9541   
host_id                                                       177932088   
host_total_listings                                                   1   

                                                         31336          31337  \
listing_id                                            36479230       36479723   
listing_name            Studio sized room with beautiful light  Room for rest   
available_days_in_year                                      16            172   
neighbourhood                                         Bushwick       Elmhurst   
neighbourhood_group                                   Brooklyn         Queens   
room_type                                         Private room   Private room   
price                                                       42             45   
minimum_nights                                               7              1   
total_reviews                                                0              0   
last_review_date                                           NaN            NaN   
reviews_per_month                                          NaN            NaN   
latitude                                              40.70418       40.74477   
longitude                                            -73.91471      -73.87727   
host_id                                               65767720       41326856   
host_total_listings                                          1              5   

                                                                    31338  \
listing_id                                                       36480292   
listing_name            Gorgeous 1.5 Bdr with a private yard- Williams...   
available_days_in_year                                                 22   
neighbourhood                                                Williamsburg   
neighbourhood_group                                              Brooklyn   
room_type                                                 Entire home/apt   
price                                                                 120   
minimum_nights                                                         20   
total_reviews                                                   

* rows seem to represent true listings without reviews
* ince we cannot evaluate limits for extremely high values in price, minimum_nights and host_total_listings from current data, these records will remain as they are

In [ ]:
# save data to use for Tableau dashboard

df.to_csv('AB_NYC_2019_clean.csv', index=False)

print("file saved")